### PR TITLE
PC-1254: Update Python version

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1254)

# Description

updates the following versions:
python 3.9.13 -> 3.11.9
django-gov-notify 0.3.0 -> 0.4.0

have checked what I can locally (various pages, csv download, phone number verification). will need checks done on dev for email sending

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable

### CODE REVIEW

File: .github/workflows/portal-checks.yml [hunk 1]
```diff
@@ -22,10 +22,10 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up Python 3.9
+      - name: Set up Python 3.11
```
Comment (samyou-softwire) [inline .github/workflows/portal-checks.yml:25]: is this actually needed? the make task will configure a docker-in-docker container anyway with correct python ver & poetry init. the Run Tests task doesn't have these steps

**Review by jdgage — CHANGES_REQUESTED** [review]

File: docker/tests/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM public.ecr.aws/docker/library/python:3.9-buster
+FROM public.ecr.aws/docker/library/python:3.11.9-bookworm
```
[the remark this answers is not in the document]
Comment (samyou-softwire) [reply docker/tests/Dockerfile:1]: I'm not easily able to update to 3.11.10 yet on my machine (company portal vers is 3.11.9), it only released on 7th September

[the remark this answers is not in the document]
Comment (jdgage) [reply docker/tests/Dockerfile:1]: Sure, fine then

File: .github/workflows/portal-checks.yml (hunk 1 above)
Comment (jdgage) [reply .github/workflows/portal-checks.yml:25]: Would you mind removing now?

**Review by jdgage — APPROVED** [review]

**Pull request merged**